docker/developers/docker-compose.selenium.yml

```diff
@@ -0,0 +1,347 @@
+  selenium-grid-contained:
+    environment: &selenium-grid-contained-environment
+      SE_EVENT_BUS_HOST: selenium
+      SE_EVENT_BUS_PUBLISH_PORT: 4442
+      SE_EVENT_BUS_SUBSCRIBE_PORT: 4443
+    <<: *selenium-grid-base
+
+  # This runs selenium-grid such that it is visible to the host machine
+  selenium-grid:
+    <<: *selenium-grid-base
+    environment: &selenium-grid-environment
+      SE_EVENT_BUS_HOST: localhost
+      SE_EVENT_BUS_PUBLISH_PORT: 4442
+      SE_EVENT_BUS_SUBSCRIBE_PORT: 4443
+    ports:
+      # The Selenium 'Publish' port
+      - "4442:4442"
+      # The Selenium 'Subscribe' port
+      - "4443:4443"
+      # The Protocol port that our testing infrastructure connects to
+      - "4444:4444"  
+
+  selenium-video:
+    hostname: selenium-video
+    image: codedotorg/selenium-video
+    build:
+      dockerfile: ./selenium-video.Dockerfile
+      context: ./selenium-video
+    environment:
+      DISPLAY_CONTAINER_NAME: code-dot-org-selenium-chrome
+    volumes:
+      - /tmp:/videos
+    networks:
+      cdo_network_test:
+
+  # A Selenium server running Chrome for running UI tests contained in Docker
+  selenium-chrome-base: &selenium-chrome-base
+    hostname: selenium
+    image: seleniarm/standalone-chromium:${SELENIUM_CHROME_VERSION}
+    environment:
+      SE_VNC_PORT: 5910
+      SE_NO_VNC_PORT: 7910
+      JAVA_OPTS: -Dwebdriver.chrome.whitelistedIps= -Dwebdriver.chrome.allowedIps=
+    shm_size: '2gb'
+    expose:
+      - "5910" # The selenium direct VNC port
+      - "4444" # The selenium router port
+      - "7910" # The selenium NoVNC service port
+    # Many default ulimits are just too high
+    # And, so, it will consume way too much memory (16GB+)
+    ulimits: *selenium-ulimits
+
+  # A Selenium server running Chrome for running UI tests locally against a
+  # containerized dashboard server.
+  selenium-chrome-contained:
+    <<: *selenium-chrome-base
+    networks:
+      cdo_network_test:
+
+  # A Selenium server running Chrome for running UI tests locally
+  selenium-chrome-native:
+    <<: *selenium-chrome-base
+    container_name: code-dot-org-selenium-chrome
+    network_mode: host
+
+  selenium-chrome:
+    image: alpine
+    depends_on:
+      selenium-chrome-native:
+        condition: service_started
+    command: >
+      /bin/sh -c "
+        echo 'Selenium Chrome Started.' &&
+        echo &&
+        echo 'To see and interact with this browser session:' &&
+        echo '- http://localhost:7910/?autoconnect=1&resize=scale&password=secret (VNC port 5910)' &&
+        echo &&
+        echo 'To shut down all selenium nodes, use:' &&
+        echo 'docker compose down selenium-chrome-native' &&
+        echo &&
+        echo 'To run a UI test with a locally running dashboard server:' &&
+        echo 'bundle exec rake test:ui browser=chrome selenium=http://localhost:4444/wd/hub feature=dashboard/test/ui/features/platform/signing_in.feature'
```

Comment:
I appreciate the cli UX nicety here, printing the HOWTO on startup like this is much more likely to get this feature used imo